yearn/contracts/SaddleStrategy.sol

```diff
@@ -0,0 +1,395 @@
+    {
+        uint256 wantBalance = balanceOfWant();
+        if (wantBalance < amountNeeded) {
+            liquidatedAmount = withdrawSome(amountNeeded.sub(wantBalance));
+            liquidatedAmount = liquidatedAmount.add(wantBalance);
+            loss = amountNeeded.sub(liquidatedAmount);
+        } else {
+            liquidatedAmount = amountNeeded;
+        }
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to liquidate everything and return the amount that got freed.
+    ///         This strategy implements the aforementioned behavior by
+    ///         withdrawing all vault's underlying tokens from the tBTC v2 Saddle
+    ///         reward pool.
+    /// @dev This function is used during emergency exit instead of prepareReturn
+    ///      to liquidate all of the strategy's positions back to the vault.
+    /// @return amountFreed Amount that got freed.
+    function liquidateAllPositions()
+        internal
+        override
+        returns (uint256 amountFreed)
+    {
+        ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).withdraw(balanceOfPool());
+
+        // Yearn docs doesn't specify clear enough what exactly should be
+        // returned here. It may be either the total balance after
+        // withdrawAllAndUnwrap or just the amount withdrawn. Currently opting
+        // for the former because of
+        // https://github.com/yearn/yearn-vaults/pull/311#discussion_r625588313.
+        // Also, usage of this result in the harvest method in the BaseStrategy
+        // seems to confirm that.
+        return balanceOfWant();
+    }
+
+    /// @notice This method is defined in the BaseStrategy contract and is meant
+    ///         to do anything necessary to prepare this strategy for migration,
+    ///         such as transferring any reserve or LP tokens, CDPs, or other
+    ///         tokens or stores of value. This strategy implements the
+    ///         aforementioned behavior by withdrawing all vault's underlying
+    ///         tokens from the tBTC v2 Saddle reward pool.
+    /// @param newStrategy Address of the new strategy meant to replace the
+    ///        current one.
+    function prepareMigration(address newStrategy) internal override {
+        // Withdraw all the LP tokens from the tBTC v2 Saddle reward pool.
+        // These tokens will be transferred to the new strategy in the `migrate`
+        // function.
+        ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).withdraw(balanceOfPool());
+        // Get all the earned KEEP tokens and transfer them to the new strategy.
+        ILPRewardsTBTCV2Saddle(lpRewardsTbtcV2Saddle).getReward();
+        IERC20(keep).safeTransfer(
+            newStrategy,
+            IERC20(keep).balanceOf(address(this))
+        );
+        //TODO: Check if both the LP Token and KEEP token should be migrated to
+        // the new strategy, or just the LP token.
```

Comment:
If we can migrate both without hassle, we should do it I think.

Reply:
Do you know if there is any risk the new strategy may be of different "type" - e.g. rewarding with a different token then KEEP?

Reply:
Good point. However, I think in that case it will be considered as a separate new strategy rather than a new version of this one.

Reply:
I agree, removed the TODO in https://github.com/keep-network/tbtc-v2/commit/802c155d8e6b78bfe17a77572814367052f3b4f7